### ISSUE TITLE
iam permission issues on remote template

### ISSUE THREAD

**Issue by sreeramsanjay** [awslabs/aws-instance-scheduler#95]
Error : 2019-06-13 - 00:31:04.734 - ERROR   : Error stopping rds instance RDS:rds-instance (Oracle EE TEST), (An error occurred (AccessDenied) when calling the StopDBInstance operation: User: arn:aws:sts::xxxxxxxx:assumed-role/instance-scheduler-t-EC2SchedulerCrossAccount-BG948RID8JGI/rds-scheduler-xxxxxxxxx is not authorized to perform: rds:StopDBInstance on resource: arn:aws:rds:us-east-1:xxxxxxxxx:snapshot:rds-ins)

**Comment by arieleeuw** [comment]
Did you create or update a new cross account role with the new template for this version? The cross account role needs additional permissions.
https://docs.aws.amazon.com/solutions/latest/instance-scheduler/templates.html

On 13 Jun 2019, at 02:45, sreeramsanjay <notifications@github.com<mailto:notifications@github.com>> wrote:


Error : 2019-06-13 - 00:31:04.734 - ERROR : Error stopping rds instance RDS:rds-instance (Oracle EE TEST), (An error occurred (AccessDenied) when calling the StopDBInstance operation: User: arn:aws:sts::xxxxxxxx:assumed-role/regn-instance-scheduler-t-EC2SchedulerCrossAccount-BG948RID8JGI/rds-scheduler-xxxxxxxxx is not authorized to perform: rds:StopDBInstance on resource: arn:aws:rds:us-east-1:xxxxxxxxx:snapshot:rds-ins)

**Comment by jolcese** [comment]
I'm having the same problem.
And the weird thing is that is trying to do a StopDBInstance on a snaphost (see arn) which makes no sense.

Another problem is that the role is missing                   
                  - 'rds:AddTagsToResource'
                  - 'rds:RemoveTagsFromResource'
for the Aurora clusters.

**Comment by georgematthew** [comment]
I am also seeing this problem in the cross-account role for version v1.2.0.

**Comment by skurlow** [comment]
I'm having a similar issue with v1.2.0. What is the fix?

2019-08-01 - 12:30:44.177 - ERROR : Error stopping rds instance RDS:iot-poc-db-prod, (An error occurred (AccessDenied) when calling the StopDBInstance operation: User: arn:aws:sts::xxxxxxxx:assumed-role/instance-scheduler-remote-EC2SchedulerCrossAccount-xxxxxxxx/rds-scheduler-xxxxxxxx is not authorized to perform: rds:StopDBInstance on resource: arn:aws:rds:ap-southeast-2:xxxxxxxx:snapshot:instancescheduler-stopped-iot-poc-db-prod)

**Comment by sreeramsanjay** [comment]
It is IAM permission issues, you need add permission for create db snapshots as well on cross account role.
"rds:CreateDBSnapshot",
"rds:CreateDBClusterSnapshot",

**Comment by FrankFioretti** [comment]
This issue is still occurring. I have added the above permissions and no change in the behavior.

**Comment by ozeebee** [comment]
Hi,
I'm having the same issue. Confirm that adding the `rds:CreateDBSnapshot` permission does not fix the issue.

I added the `rds:StopDBInstance` permissions on the `arn:aws:rds:*:123456789012:snapshot:*` resource and I moved the `rds:DescribeDBSnapshots` permission from the `arn:aws:rds:*:123456789012:db:*` to the `*` resource. 

So my policy looks like this:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Action": [
                "rds:DeleteDBSnapshot",
                "rds:StopDBInstance"
            ],
            "Resource": "arn:aws:rds:*:123456789012:snapshot:*",
            "Effect": "Allow"
        },
        {
            "Action": [
                "rds:AddTagsToResource",
                "rds:RemoveTagsFromResource",
                "rds:StartDBInstance",
                "rds:StopDBInstance"
            ],
            "Resource": "arn:aws:rds:*:123456789012:db:*",
            "Effect": "Allow"
        },
        {
            "Action": [
                "rds:StartDBCluster",
                "rds:StopDBCluster"
            ],
            "Resource": [
                "arn:aws:rds:*:123456789012:cluster:*"
            ],
            "Effect": "Allow"
        },
        {
            "Action": [
                "ec2:StartInstances",
                "ec2:StopInstances",
                "ec2:CreateTags",
                "ec2:DeleteTags"
            ],
            "Resource": [
                "arn:aws:ec2:*:123456789012:instance/*"
            ],
            "Effect": "Allow"
        },
        {
            "Action": [
                "rds:DescribeDBClusters",
                "rds:DescribeDBInstances",
                "rds:DescribeDBSnapshots",
                "ec2:DescribeInstances",
                "ec2:DescribeRegions",
                "ec2:ModifyInstanceAttribute",
                "ssm:DescribeMaintenanceWindows",
                "tag:GetResources"
            ],
            "Resource": [
                "*"
            ],
            "Effect": "Allow"
        }
    ]
}
```

I am not an IAM specialist but it looks like some items in this policy are redundant (like the `StopDBInstance`)
However, now it seems to work (to be confirmed). 

Still generating an error when trying to delete a non-existent snapshot the first time it stops the DB though.

**Comment by tapughose** [comment]
this has been covered by [this PR](https://github.com/awslabs/aws-instance-scheduler/pull/107) and will be included in upcoming release (v1.3).

**Comment by georgebearden** [comment]
This issue should be resolved in the latest release.
